clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -344,10 +352,17 @@ class ComplexExprEmitter
     return QualType();
   }
 
+  template <typename T>
+  FPOptionsOverride getStoredFPFeaturesOrDefault(const T *E,
+                                                 const CodeGenFunction &CGF) {
```

Comment:
`CGF` is unused, can it be removed?

Why a template type? Everything is passing in some subclass of `Expr` it seems.

Also, the function should be marked `static`, but why not add it as a member function alongside `getStoredFPFeatures()`?

Reply:
`Expr` doesn't have a `hasStoredFPFeatures` or `getSoreFpFeatures`.